src/v/net/server.cc

```diff
@@ -264,7 +264,8 @@ server::accept_finish(ss::sstring name, ss::future<ss::accept_result> f_cs_sa) {
       this->name(),
       ar.remote_address,
       name);
-    if (_conn_gate.is_closed()) {
+    if (_as.abort_requested() || _conn_gate.is_closed()) {
```

Comment:
nice find.. fix looks fine to me but a bit hacky...wondering if the right way to fix this is to 

1. to stop accepting, 
2. wait until we drain the inflight accepts 
3. initiate a shutdown_input.

I think we are just missing (2)?

Reply:
good point @bharathv you said what i couldn't figure out how to say...

Reply:
@bharathv added (2) as you suggested. Was hesitant as this requires making `shutdown_input` async.

Reply:
An alternative solution I thought of was to use an rwlock for connections. This one seems a bit cleaner though.

Reply:
>@bharathv added [(2)](https://github.com/redpanda-data/redpanda/pull/13739/commits/1f4dbb971581bd366c5763e36a4c03f21580d8dc) as you suggested. Was hesitant as this requires making shutdown_input async.

One argument for not making shutdown_input async is that, frankly, this is all pretty old and fragile code. Perhaps we shouldn't change its scheduling behavior too much?

Reply:
IMO the latest commit seems fairly self-contained, low risk since its exercised at shutdown and has pretty good test coverage given every test exercises it.